src/in-depth/signals.md

```diff
@@ -1 +1,125 @@
 # Signal handling
+
+Processes
+like command line applications
+need to react to signals sent by the operating system.
+The most common example is probably <kbd>Ctrl</kbd>+<kbd>C</kbd>,
+the signal that typically tells a process to terminate.
+To handle signals in Rust programs
+you need to consider how you can receive these signals
+as well as how you can react to them.
+
+<aside>
+
+**Note:**
+If your applications does not require special cleanup,
+the default handling is fine
+(i.e. exit immediately and let the OS cleanup).
+In that case:
+No need to do what this chapter tells you!
+For applications that need to clean up after themselves,
+for example to correct close network connections,
+remove temporary files,
+or reset system settings,
+this chapter is however very relevant!
+
+</aside>
+
+## Differences between operating systems
+
+On Unix systems
+(like Linux, macOS, and FreeBSD)
+a process can receive [signals]
+It can either react to them
+in a default (OS-provided) way,
+catch the signal and handle them in a program-defined way,
+or ignore the signal entirely.
+
+[signals]: https://manpages.ubuntu.com/manpages/bionic/en/man7/signal.7.html
+[signal-safety man page]: http://manpages.ubuntu.com/manpages/bionic/man7/signal-safety.7.html
+
+Windows does not have signals.
+You can use [Console Handlers]
+to define callbacks that get executed when an event occurs.
+There is also [structured exception handling]
+which handles all the various types of system exceptions such as division by zero, invalid access exceptions, stack overflow, and so on
+
+[Console Handlers]: https://docs.microsoft.com/de-de/windows/console/console-control-handlers
+[structured exception handling]: https://docs.microsoft.com/en-us/windows/desktop/debug/structured-exception-handling
+
+## First off: Handling Ctrl+C
+
+The [ctrlc] crate does just what the name suggests:
+It allows you to react to the user pressing <kbd>Ctrl</kbd>+<kbd>C</kbd>,
+in a cross-platform way.
+The main way to use the crate is this:
+
+[ctrlc]: https://crates.io/crates/ctrlc
+
+```rust,ignore
+{{#import signals-ctrlc.rs:1:7}}
+```
+
+This is, of course, not that helpful:
+It only prints a message but otherwise doesn't stop the program.
+
+In a real-world program,
+it's a good idea to instead set a variable in the signal handler
+that you then check in various places in your program.
+For example,
+you can set an `Arc<AtomicBool>`
+(a boolean that shareable between threads)
```

Comment:
"that is shareable"? Or you could reword this as "a boolean shared between threads".